src/linear_algebra/finite_dimensional.lean

```diff
@@ -950,6 +950,10 @@ theorem finrank_range_add_finrank_ker [finite_dimensional K V] (f : V →ₗ[K]
   finrank K f.range + finrank K f.ker = finrank K V :=
 by { rw [← f.quot_ker_equiv_range.finrank_eq], exact submodule.finrank_quotient_add_finrank _ }
 
+/-- The dimensions of the domain and range of an injective linear map are equal. -/
+lemma finrank_range_of_inj {f : V →ₗ[K] V₂} [finite_dimensional K V] (hf: function.injective f) :
```

Comment:
You shouldn't need finite dimensionality here. First state it for the rank, and then deduce it for the finrank.

Reply:
You're right that this doesn't require finite dimensionality, but that's the case I'm after.  It is stated more generally for the rank here:

https://leanprover-community.github.io/mathlib_docs/linear_algebra/dimension.html#dim_eq_of_injective

I ran into an issue where trying to state `module.rank K V = module.rank K f.range` tells me there is a type mismatch (because V and f.range have different types).  I notice that in `dim_eq_of_injective`, the corresponding modules `M` and `M_1` are built with the same type.  I'm not sure why.

Can you offer some guidance on how to use `dim_eq_of_injective` to deduce the finite dimensional result above?

Reply:
I've pushed a fix that removes the useless assumption in previous lemmas, so that your proof goes through without a finite-dimensionality assumption.

Reply:
Oh, I think I better understand your suggestion now.  Thanks for the fix!

I appreciate your help.